### ISSUE TITLE
How to ignore certain cases?

### ISSUE THREAD

**Issue by benjick** [JacobNWolf/biome-unnecessary-effect#6]
Hello!

Awesome plugin. I have some places where I want to keep `useEffect` though (I think), but this plugin says I shouldn't, like this:

```tsx
useEffect(() => {
  return editor.registerNodeTransform(ParagraphNode, (node) => {
    if (node.getTextContent().trim() === "") {
      node.remove();
    }
  });
}, [editor]);
```

Can I disable it for that useEffect somehow?

**Comment by JacobNWolf** [comment]
Thanks for reporting! 

This is a valid effect — it's a cleanup function. It was being false positive flagged by the `avoidEventHandler` rule. I've patched this in #9, which will be in the next release.

For documentation purposes and to answer your original question, you can do the following above the `useEffect` itself to ignore the rule: 

```typescript
// biome-ignore lint: <explanation>
```

I've added [a test case](https://github.com/JacobNWolf/biome-unnecessary-effect/blob/5032554ecb616c70754015eac428c0b88ec1813a/tests/fixtures/suppression.tsx) to demonstrate in the PR also.

**Comment by JacobNWolf** [comment]
Ok, this has been [released](https://www.npmjs.com/package/@jacobwolf/biome-unnecessary-effect)! Thanks again for flagging.

**Comment by benjick** [comment]
Thanks a lot!

What about this case when `messages` is just a prop?

```ts
  useEffect(() => {
    if (messagesContainerRef.current) {
      messagesContainerRef.current.scrollTop =
        messagesContainerRef.current.scrollHeight;
    }
  }, [messages]);
```

Do you know if there's a more specific lint suppression than using `// biome-ignore lint: <explanation>`? I've tried googling but I couldn't find any. Would be nice to not turn off every rule.
